### DISCUSSION TITLE
“Bet was not accepted within the expected timeframe”

### DISCUSSION THREAD

**Discussion by Schwenckejr** [cloudbet/docs#67] [General]
Hello, 

Im often having issues placing bets with the API, as I’m receiving the error message:

“Bet was not accepted within the expected timeframe” 

And i would verry much like to know if there is anything i can do to sort out this problem?

The same matches I’m able to place the bets manually, so it’s not a liquidity issue. 

Regards Thomas

**Comment by kgravenreuth** [comment]
Hi @Schwenckejr ,

Could you please give us more details, such as a reference bet request payload, response payload and http status so that we can investigate further?

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by Schwenckejr** [reply]
Here is what I am are facing. 

When I place a bet we check for maxStake via the data from endpoint /v2/odds/events/${id}

I make sure that the stake amount is less than maxStake when placing a bet, yet I get the following error: 

 Error: {"referenceId":"<REF_ID>","price":"3.75","eventId":"19573300","marketUrl":"tennis.winner/away","side":"BACK","currency":"BTC","stake":"0.0014415","createTime":"2023-08-17T15:04:52Z","status":"STAKE_ABOVE_MAX","returnAmount":"0.0","eventName":"","sportsKey":"tennis","competitionId":"11335","categoryKey":"itf_women","customerReference":"<CUSTOMER_REF>","error":""}. 

When I get this error, I make a retry by fetching the latest information about the event again, and adjust stake amount to be less than maxStake, yet the error keeps happening. This only happens 20% of the time when placing bets. 

Any recommendations? I have tried adjusting the stake amount to be 10% less on the frist retry but I run into the same error most of the time.

**Reply by kgravenreuth** [reply]
Thanks for the details @Schwenckejr 

After review, this could be something specific to your account. Could you please email me at `klaus @ cloudbet.com` with your Cloudbet username so that we can follow up further in more detail? Thanks!

Regards,
Klaus Gravenreuth
Product & Engineering @ Cloudbet

**Reply by kgravenreuth** [reply]
@Schwenckejr 

After review, this is a general logic where the `maxStake` in the `Feed API` response is then used in the `Trading API` place bet request. Please make sure that the difference in time between these two calls is minimised so that you get the latest value.

Furthermore, please use the `stake` in the Trading API place bet error response for the latest information about the new max stake value which you can use to place a bet again. This is mentioned in our [Trading API documentation](https://www.cloudbet.com/api/?urls.primaryName=Trading#/Trading/PlaceBet):


<img width="1319" alt="place_bet_response" src="https://github.com/Cloudbet/docs/assets/21211077/eb57f083-8adf-4ace-9232-863dbdc19bd9">

Please let me know if this clears up your doubts or if you still have any questions.

**Comment by Schwenckejr** [comment]
Can you elaborate on the the "stake" property in the place bet error response? Is that the adjusted amount that is allowed to be placed? The documentation doesn't mention what it does.

Secondly, the STAKE_ABOVE_MAX error never happens in the place bet error response. It happens when we check the status of the bet placed in the GET bet status endpoint, as the status in the response of the place bet endpoint is mostly PENDING_ACCEPTANCE.

Sent from my iPhone

On Aug 22, 2023, at 10:45 AM, Klaus Gravenreuth ***@***.***> wrote:

﻿

@Schwenckejr<https://github.com/Schwenckejr>

After review, this is a general logic where the maxStake in the Feed API response is then used in the Trading API place bet request. Please make sure that the difference in time between these two calls is minimised so that you get the latest value.

Furthermore, please use the stake in the Trading API place bet error response for the latest information about the new max stake value which you can use to place a bet again. This is mentioned in our Trading API documentation<https://stg.cloudbet.com/api/?urls.primaryName=Trading#/Trading/PlaceBet>:

[place_bet_response]<https://user-images.githubusercontent.com/21211077/262290811-eb57f083-8adf-4ace-9232-863dbdc19bd9.png>

Please let me know if this clears up your doubts or if you still have any questions.

**Reply by kgravenreuth** [reply]
> Can you elaborate on the the "stake" property in the place bet error response? Is that the adjusted amount that is allowed to be placed?

Yes, that's right. It is mentioned in the documentation as part of the `status` response for certain status codes (as highlighted in my screenshot above):

```
stake: Stake in numeric format for given currency
```

```
STAKE_ABOVE_MAX - stake requested was above the current maximum stake on a selection. Please reference `stake` from response payload about the corrected value for retry
STAKE_BELOW_MIN - stake requested was below the current minimum stake on a selection. Please reference `stake` from response payload about the corrected value for retry
LIABILITY_LIMIT_EXCEEDED - your current liability limit on this event was exceeded. Please reference `stake` from response payload about the corrected value for retry
```

> Secondly, the STAKE_ABOVE_MAX error never happens in the place bet error response. It happens when we check the status of the bet placed in the GET bet status endpoint, as the status in the response of the place bet endpoint is mostly PENDING_ACCEPTANCE

Yes, this could happen as the bets might not be accepted by our system immediately and require you to poll the bet status endpoint afterward. The same logic applies for the `stake` field in both these endpoints.

**Comment by Schwenckejr** [comment]
Hello Klaus

Thank you for your responses, I’m pretty sure I’ve worked it out now! At least no errors have been happening the past day, so fingers crossed.

Regards Thomas

Sent from my iPhone

On Aug 25, 2023, at 5:48 AM, Klaus Gravenreuth ***@***.***> wrote:

﻿

Can you elaborate on the the "stake" property in the place bet error response? Is that the adjusted amount that is allowed to be placed?

Yes, that's right. It is mentioned in the documentation as part of the status response for certain status codes (as highlighted in my screenshot above):

stake: Stake in numeric format for given currency


STAKE_ABOVE_MAX - stake requested was above the current maximum stake on a selection. Please reference `stake` from response payload about the corrected value for retry
STAKE_BELOW_MIN - stake requested was below the current minimum stake on a selection. Please reference `stake` from response payload about the corrected value for retry
LIABILITY_LIMIT_EXCEEDED - your current liability limit on this event was exceeded. Please reference `stake` from response payload about the corrected value for retry


Secondly, the STAKE_ABOVE_MAX error never happens in the place bet error response. It happens when we check the status of the bet placed in the GET bet status endpoint, as the status in the response of the place bet endpoint is mostly PENDING_ACCEPTANCE

Yes, this could happen as the bets might not be accepted by our system immediately and require you to poll the bet status endpoint afterward. The same logic applies for the stake field in both these endpoints.
